### ISSUE TITLE
Implement Google error model

### ISSUE THREAD

**Issue by davidbnk** [grpc/grpc-node#1716]
Any plans on implementing [Google error model](https://cloud.google.com/apis/design/errors#error_model)?

The current model only supports status code and a message
https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/call-stream.ts#L55

**Comment by murgatroid99** [comment]
Servers that provide those details put them in the `grpc-error-details-bin` header. I suggest using the [node-grpc-error-details](https://github.com/stackpath/node-grpc-error-details) project to parse that header.

**Comment by davidbnk** [comment]
> 
> 
> Servers that provide those details put them in the `grpc-error-details-bin` header. I suggest using the [node-grpc-error-details](https://github.com/stackpath/node-grpc-error-details) project to parse that header.

I'm using grpc-node as a server and consuming the server from a dart client. I've tried to use the metadata field to send details like
```
metadata.set('key1', 'value1');
```
But in the client the details array is empty. Is this the right way?

**Comment by murgatroid99** [comment]
No. The way this works is that the error details are represented as a `google.rpc.Status` protobuf message, which is serialized and then included in the trailing metadata, using the `grpc-error-details-bin` metadata key.

If you don't need arbitrary objects in the error details, and you just want string key/value pairs, this Google error model is overkill. In that case, you can just set metadata keys and values as you mentioned, and then access them on the client in the `metadata` property of the status or error object.
